### PR TITLE
Fix azure backup writing multipart blocks by 1mb (read buffer size) instead of `max_upload_part_size` (in non-native copy case)

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix azure backup writing multipart blocks as 1mb (read buffer size) instead of max_upload_part_size.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [x]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [x]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

</details>

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 93b11c57a08504c60d57806955cae0764f95c933 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64117/93b11c57a08504c60d57806955cae0764f95c933/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by alexey-milovidov — APPROVED** [review]

**Review by alesapin — APPROVED** [review]

File: src/IO/AzureBlobStorage/copyAzureBlobStorageFile.cpp [hunk 1]
```diff
@@ -94,11 +95,56 @@ namespace
...
+                String msg;
+                if (num_parts < 1)
+                    msg = "Number of parts is zero";
+                else if (num_parts > max_part_number)
+                    msg = fmt::format("Number of parts exceeds {}", num_parts, max_part_number);
+                else if (part_size < min_upload_part_size)
+                    msg = fmt::format("Size of a part is less than {}", part_size, min_upload_part_size);
+                else
+                    msg = fmt::format("Size of a part exceeds {}", part_size, max_upload_part_size);
+
```
Comment (antonio2368) [inline src/IO/AzureBlobStorage/copyAzureBlobStorageFile.cpp:138]: you use only 1 `{}` while having 2 variables?

File: src/IO/AzureBlobStorage/copyAzureBlobStorageFile.cpp [hunk 2]
```diff
@@ -94,11 +95,56 @@ namespace
...
+                throw Exception(
+                    ErrorCodes::INVALID_CONFIG_PARAMETER,
+                    "{} while writing {} bytes to Azure. Check max_part_number = {}, "
+                    "min_upload_part_size = {}, max_upload_part_size = {}",
+                    msg, total_size, max_part_number, min_upload_part_size, max_upload_part_size);
+            }
```
Comment (antonio2368) [inline src/IO/AzureBlobStorage/copyAzureBlobStorageFile.cpp:144]: you repeat checks, why not have something like:
```cpp
String msg;
if (bad1)
   msg = "bad1";
else if (bad2)
...

if (msg)
   throw msg
```

**Review by antonio2368 — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (tsan) [1/6] — fail: 1

Not related to changes:

`test_modify_engine_on_restart/test_zk_path.py::test_modify_engine_fails_if_zk_path_exists`

` Exception: Command ['docker', 'exec', '-u', 'root', 'roottestmodifyengineonrestartzkpath_ch1_1', 'bash', '-c', "gdb -batch -ex 'thread apply all bt full' -p 748"] return non-zero code 1: Recursive internal problem.`

**Pull request merged**